### ISSUE TITLE
ClusterAdmin swallows TopicError ErrMsg

### ISSUE THREAD

**Issue by dnwe** [Shopify/sarama#1153]
##### Versions

Sarama Version: e7238b119b7daab993720f0153eafb88e2b0ac1f
Kafka Version: 1.1.1
Go Version: go1.10

##### Configuration

N/A

##### Logs

N/A

##### Problem Description

The `ClusterAdmin` interface added under https://github.com/Shopify/sarama/pull/1055 provides an administrative client for Kafka similar to the official Java client based on design discussions under https://github.com/Shopify/sarama/issues/1048

However, in the case of a `TopicError` being returned by one of the calls, only the underlying `KError` is returned to the user as it satisfies the built-in `error` interface.

https://github.com/Shopify/sarama/blob/e7238b119b7daab993720f0153eafb88e2b0ac1f/admin.go#L141-L148

However, in the case of (e.g.,) `sarama.ErrPolicyViolation`, `sarama.ErrInvalidPartitions`, `sarama.ErrInvalidReplicaAssignment` etc. there may well be a specific error message in the `TopicError` `ErrMsg` that is being hidden from the caller.

Shouldn't `TopicError` itself be extended to satisfy the `error` interface by appending any `ErrMsg` to the generic one?

e.g.,
```go
type TopicError struct {
    Err    KError
    ErrMsg *string
}

func (t *TopicError) Error() string {
	text := t.Err.Error()
	if t.ErrMsg != nil {
		text = fmt.Sprintf("%s - %s", text, *t.ErrMsg)
	}
	return text
}
```

**Comment by dnwe** [comment]
Returning the full TopicError would allow type inference to extract the kError and errMsg separely if that's appropriate too

**Comment by varun06** [comment]
@dnwe This should be fixed now right?

**Comment by dnwe** [comment]
Yep. Fixed be #1154
